readjson.go

```diff
@@ -89,23 +104,43 @@ func readSMARTctlDevices(logger log.Logger) gjson.Result {
 	return parseJSON(string(out))
 }
 
+// Refresh all devices' json
+func refreshAllDevices(logger log.Logger, devices []Device) {
+	if *smartctlFakeData {
+		return
+	}
+
+	pool := createPool()
```

Comment:
This seems to be implementing a [`sync.WaitGroup`](https://pkg.go.dev/sync#WaitGroup). Or maybe better, use [`errgroup.Group`](https://pkg.go.dev/golang.org/x/sync/errgroup).

Reply:
Thanks. Will try to refactor it later, after the fate of #205 is decided.

Reply:
> Thanks. Will try to refactor it later, after the fate of #205 is decided.

Merged. Can you fix the conflicts for merge?